modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -1120,6 +1122,36 @@ CV__DNN_INLINE_NS_BEGIN
      */
     CV_EXPORTS_W Net readNetFromONNX(const std::vector<uchar>& buffer);
 
+     /** @brief This struct contains all the configuration of TensorRT backend.
+      */
+    struct CV_EXPORTS_W_SIMPLE TrtConfig
+    {
+        CV_PROP_RW int deviceId;      //!< Running TensorRT with specific GPU
+        CV_PROP_RW bool useCache;     //!< It will speed up the ONNX model initialization. Write back the converted trt file to where the input model is located.
+        CV_PROP_RW String cachePath;   //!< Setting the path to write back the converted .trt file to specific path instead of the input file location.
+        CV_PROP_RW bool useFP16;      //!< Inference model with FP16, need hardware supports FP16, it may affect the accuracy.
+        CV_PROP_RW bool useTimeCache; //!< To speed up the ONNX -> trt converting processing, it will write the time cache file to cachePath.
+        CV_PROP_RW String inputName;            //!< Setting the input name and shape to convert dynamic shape ONNX to trt. Current only support single input.
```

Comment:
Do we really need input name for ONNX conversion or there is a generated one? We can always change it by `setInputNames` later.

Reply:
Hi, this optional is set for models with dynamic shapes. And the TensorRT converter needs input name and specific static shape to convert the dynamic to static input node. The converter processing happens at the `readNet` function.